src/utils/secretmanager.rs

```diff
@@ -0,0 +1,66 @@
+use async_trait::async_trait;
+use aws_config::{BehaviorVersion, Region};
+use aws_sdk_secretsmanager::Client as awsClient;
+
+use super::errors::Error;
+pub struct Client {
+    secret_name: String,
+    _region: Region,
+    aws_client: awsClient,
+}
+
+pub struct Secret {
+    secret_name: String,
+    secret_value: String,
+}
+
+#[async_trait]
+pub trait Operations {
+    async fn get_key(&self) -> Result<Option<String>, Error>;
+
+    async fn store_key(&self, secret: Secret) -> Result<(), Error>;
+}
+
+impl Client {
+    pub async fn new(secret_name: String, _region: Region) -> Self {
+        let config = aws_config::defaults(BehaviorVersion::v2025_01_17())
+            .region(_region.clone())
+            .load()
+            .await;
+        let aws_client = aws_sdk_secretsmanager::Client::new(&config);
```

Comment:
The `AwsClient` type is already imported. Use it directly.
```suggestion
        let aws_client = AwsClient::new(&config);
```